nexus/tests/integration_tests/metrics.rs

```diff
@@ -5,64 +5,35 @@
 use dropshot::test_util::ClientTestContext;
 use dropshot::ResultsPage;
 use nexus_test_utils::resource_helpers::objects_list_page_authz;
-use omicron_common::backoff;
 use oximeter::types::Datum;
 use oximeter::types::Measurement;
 
 pub async fn query_for_metrics_until_they_exist(
     client: &ClientTestContext,
     path: &str,
 ) -> ResultsPage<Measurement> {
-    backoff::retry_notify(
-        backoff::retry_policy_local(),
-        || async {
-            let measurements: ResultsPage<Measurement> =
-                objects_list_page_authz(client, path).await;
+    loop {
```

Comment:
`wait_for_condition` basically does this I think (no need to use it though -- it's simple enough)

(sorry I wasn't more explicit about that before)